code/modules/cards/cardhand.dm

```diff
@@ -6,7 +6,7 @@
 	w_class = WEIGHT_CLASS_TINY
 	worn_icon_state = "card"
 
-/obj/item/toy/cards/cardhand/Initialize(mapload, list/cards_to_combine)
+/obj/item/toy/cards/cardhand/Initialize(mapload, list/cards_to_combine = list())
 	. = ..()
 
 	if(!LAZYLEN(cards_to_combine) && (mapload && !LAZYLEN(cards)))
```

Comment:
This is an error state no? why cover it up with the default value if you want to error if no cards are passed in?

Reply:
Yes it is used for error checking.  There are two situations where we want cardhands to spawn:

- Mapped in cardhands (that can't use the `cards_to_combine` arg so they use the `cards` VV instead.
- Cardhands created during runtime when combining cards together. (then we want only `cards_to_combine`)

The reason we are making the arg default value be a list is because it is a null by default and we need to be able to add singlecards to the cards_to_combine list.  If it's null, we can't do list operations.

Reply:
why is it a lazylist if null isn't a possible input exactly?

Reply:
We are checking if the list is empty, if it is then it crashes.  If the list is null it will runtime since two singlecards will attempt to `+=` join together like it's a list. (when it's not a list)  So we always want to have the arg default as a list instead of null.

Reply:
Are you satisfied with this explanation?

- Mapped cardhands use the `cards` VV
- Runtime spawned cardhands use the `cards_to_combine` arg

Reply:
I do not understand why you're treating cards_to_combine as a lazylist if it's never going to be null.
Why is it not better to make a new list to store the new cards in, rather then trying to reuse an old one that represents something different?

Reply:
> I do not understand why you're treating cards_to_combine as a lazylist if it's never going to be null.

We are using `LAZYLEN(cards_to_combine)` to check if the length is zero.  We could just use `length(card_to_combine)` since it does the same thing.  The lazylist procs are just extra tooling for normal lists that are optional.

> Why is it not better to make a new list to store the new cards in, rather then trying to reuse an old one that represents something different?

So I think what you are asking here is, why don't I have two lists, one for `runtime_spawned_cardhands` and another for `mapped_spawned_cardhands`?

To generate both cardhands in either scenario requires you to spawn them via looping through a list.  We have this list already formatted correctly for `runtime_spawned_cardhands` so we just need to loop through it.  For `mapped_spawned_cardhands` we need to transfer the VV cards into the list manually.  They both can use the same list.  That list is `cards_to_combine` since we don't care _how_ the cards get added to the list.

I'll try to refactor the Initialization code so it's more human legible.  I've noticed a few times where people have avoided using `Initialization()` and wrote their own `generateObject()` procs in the codebase.  I think this is due to the lack of examples using `mapload` var and `loc` args properly.

Reply:
Refactored with 8dc345a25cb626364e8a0b5cb2171255452b6b4d